### PR TITLE
BUG: random.beta with small parameters

### PR DESCRIPTION
**Pull request by jaimefrio**

### CODE REVIEW

File: numpy/random/mtrand/distributions.c [hunk 1]
```diff
@@ -199,7 +199,20 @@ double rk_beta(rk_state *state, double a, double b)
 
             if ((X + Y) <= 1.0)
             {
-                return X / (X + Y);
+                if (X +Y > 0)
+                {
```
Comment (charris) [inline numpy/random/mtrand/distributions.c]: `{` on previous line.

Comment (charris) [inline numpy/random/mtrand/distributions.c]: OK, IIRC, this is random.c style ;)

File: numpy/random/mtrand/distributions.c [hunk 2]
```diff
@@ -199,7 +199,20 @@ double rk_beta(rk_state *state, double a, double b)
...
+                    return X / (X + Y);
+                }
+                else
+                {
+                    double logX = log(U) / a;
+                    double logY = log(V) / b;
+                    double logM = logX > logY ? logX : logY;
+                    logX -=logM;
```
Comment (charris) [inline numpy/random/mtrand/distributions.c]: Space after `-=`.

**Comment by charris** [comment]
Minor style nitpick. LGTM otherwise, assuming the code is doing the right thing.

**Comment by jaimefrio** [comment]
Fixed the missing space, left the braces to match the surrounding code.

**Comment by charris** [comment]
Thanks Jaime.

**Pull request merged**